### ISSUE TITLE
[Q] How to create ejabberd certs? 

### ISSUE THREAD

**Issue by soakes** [Neilpang/acme.sh#829]
Hi Guys,

This is more of a general question as I am in the middle of changing my SSL certs which is used with my jabber server to use letsencrypt.

Now if your not aware, ejabberd requires the pem file to be created in the following order..

- private key
- certificate
- chains

Now the problem is, the fullchain doesn't include the private key, so my question is, how are people adding the private key each time a new cert is generated?

Currently I am using DNS method using the following command..

```
acme.sh --issue --dns -d jabber.example.co.uk \
--certpath      /etc/ejabberd/certs/jabber.example.co.uk-cert  \
--keypath       /etc/ejabberd/certs/jabber.example.co.uk-key.pem  \
--fullchainpath /etc/ejabberd/certs/jabber.example.co.uk-fullchain.pem \
--reloadcmd     "service ejabberd restart"
```

I guess I could cat the private key file to the top of the file before restart on the reloadcmd but that seems a little messy.  Does anyone have a better way to do this?

Thanks in advance.

**Comment by Neilpang** [comment]
1. Do not use `--dns` mode, it can not be renewed automatically.   Please use dns api mode or other mode instead.

1. Yes, using reloadcmd to conbine the key and cert is the right choice.

**Comment by soakes** [comment]
REF: dns mode, I didn't know that and will change that right away. Thank you.

Any suggestions with mode to use as my DNS cluster is all done via bind.

**Comment by cthart** [comment]
Not sure how soakes solved this problem, but ejabberd (I'm using 18.04) does allow the specification of multiple "certfiles" in the ejabberd.yml The comment there even says "List all available PEM files containing certificates for your domains, chains of certificates or certificate keys. Full chains will be built automatically by ejabberd."
I have listed the key file and the fullchain file there and it works just fine. I did need to change ownership of the files to ejabberd:ejabberd Hopefully that ownership will be retained when acme.sh updates the files in 2 months time.
